File: 2. Bubble Sort/BubbleSort.c
#include <stdio.h>
#include <stdlib.h>

void swap(int *num1, int *num2){
    int temp = *num1;
    *num1 = *num2;
    *num2 = temp;
}


void BubbleSort(int arr[], int n){
    int swapped = 0 ;
    for(int i=0;i<n-1;i++){
        for(int j=0;j<n-1-i;j++){
            if(arr[j]>arr[j+1]){
                swap(&arr[j],&arr[j+1]);
                swapped=1;
            }
        }
        if(swapped == 0){
            break;
        }
    }
}

int main(){
    int arr[6] = {9,8,6,5,3,2};
    int size = 6;

    printf("Array before Sorted :\n");
    for (int i=0;i<size;i++){
        printf("%d ",arr[i]);
    }
    printf("\n");
    BubbleSort(arr,size);
    printf("Array after sorted :\n");
    
    for (int i=0;i<size;i++){
        printf("%d ",arr[i]);
    }
}

File: 2. Bubble Sort/BubbleSort.py
def BubbleSort(arr):
    n = len(arr)
    swapped = False
    for i in range(n-1):
        for j in range(n-1-i):
            if(arr[j]>arr[j+1]):
                temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = temp
                swapped = True
        if(swapped == False):
            break

if __name__ == "__main__":
    arr = [9,8,7,6,5,4,3,2,1]
    print("Array before sorted... :",arr)
    BubbleSort(arr)
    print("Array after sorted... :", arr)
